ivi-layermanagement-api/ilmControl/src/ilm_control_wayland_platform.c

```diff
@@ -2136,11 +2138,20 @@ static void screenshot_done(void *data, struct ivi_screenshot *ivi_screenshot,
 
     munmap(buffer, size);
 
-    if (save_as_bitmap(filename, (const char *)image_buffer,
-                       image_size, width, height, bytes_per_pixel * 8) == 0) {
-        ctx_scrshot->result = ILM_SUCCESS;
+    if ((filename_ext = strstr(filename, ".png")) && (strlen(filename_ext) == 4)) {
+        if (save_as_png(filename, (const char *)image_buffer,
+                              width, height, bytes_per_pixel * 8, has_alpha) == 0) {
+            ctx_scrshot->result = ILM_SUCCESS;
+        } else {
+            fprintf(stderr, "failed to write screenshot as png file: %m\n");
+        }
     } else {
```

Comment:
Would it make sense to check the file extension for bmp as well, just in case the user specifies any other unsupported format like `.jpeg`?

Reply:
@efriedrich `jpeg` extension is not supported according to the code :) I agree, it makes sense to keep the old behavior, i.e. use bmp format if no extension (or unsupported extension) specified, but the user has to be warned if the file extension does not exist or does not match the file format! We had a lot of troubles recently when a user tried to create "png" screenshot and open it with a desktop image viewer. There was no error message and no user friendly way to find the root cause!

Reply:
@michaelrodin I forgot to add the warning message. I will add it.